File: 11-ContainerWithMostWater/11-ContainerWithMostWater.go
// Last updated: 18/12/2025, 20:21:33
func maxArea(height []int) int {
    hl := len(height)
    i := 0
    j := hl-1
    max_area := 0
    for i<j && i<hl && j > -1 {
        width := j-i
        curr_area :=0
        if height[i] > height[j]{
            curr_area = height[j] * width
            j = j - 1
        } else {
            curr_area = height[i] * width
            i = i  + 1
        }
        if curr_area > max_area {
            max_area = curr_area
        }

    }
    return max_area



    // max_area := 0
    // // h÷l2 := int(hl /2)
    // for i < hl - 1{
    //     j = i + 1
    //     for j< hl{
    //         smallheight := height[i]
    //         smallheight := math.Min(height[i] , height[j])
    //         if height[i] > height[j] {
    //             smallheight = height[j]
    //         }
    //         curr_area := smallheight * (j- i) 

    //         if curr_area > max_area {
    //             max_area = curr_area
    //         }
    //         j +=1
    //     }

    //     i+=1

    // }
    // return max_area
    
}

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 18/12/2025, 20:21:28
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxwater = 0
        n = len(height)
        left = 0
        right = n -1
        while left < right:
            width = right - left
            
            minlen = min(height[left], height[right])
            area = width * minlen
            maxwater = max(area, maxwater)

            if height[left] < height[right]:
                left +=1
            else:
                right -=1
        return maxwater
        